ui/utils.py

```diff
@@ -170,3 +177,119 @@ def write_x509_files():
     """Write the x509 certificate and key to files"""
     write_to_file(settings.MIT_WS_CERTIFICATE_FILE, settings.MIT_WS_CERTIFICATE)
     write_to_file(settings.MIT_WS_PRIVATE_KEY_FILE, settings.MIT_WS_PRIVATE_KEY)
+
+
+def get_video_analytics(video_key):
+    """Get video analytics data from Google Analytics."""
+    ga_client = get_google_analytics_client()
+    ga_response = ga_client.reports().batchGet(
+        body=generate_google_analytics_query(video_key)).execute()
+    try:
+        return parse_google_analytics_response(ga_response)
+    except Exception as exc:
+        raise GoogleAnalyticsException(
+            'Could not parse analytics response') from exc
+
+
+def get_google_analytics_client():
+    """Gets a Google Analytics client.
+
+    Returns:
+        analytics_client: An analytics client
+    """
+    _check_files_exist([settings.GA_KEYFILE])
```

Comment:
The MIT certificate and private key files are generated on app startup [here](https://github.com/mitodl/odl-video-service/blob/6beb8c88b35247215f88f1f3312a18263285a825/ui/apps.py#L27).  The same would need to be done for the GA keyfile, with `settings.GA_KEYFILE` equaling the contents of the file (same as `MIT_WS_CERTIFICATE` and `MIT_WS_PRIVATE_KEY`), but in that case you can skip the file creation and use the method `ServiceAccountCredentials.from_json_keyfile_dict` instead of `ServiceAccountCredentials.from_json_keyfile_name`), something like this:

```
try:
    credentials = ServiceAccountCredentials.from_json_keyfile_dict(json.loads(settings.GA_KEYFILE))
```